google_compute_engine/network_utils.py

```diff
@@ -38,11 +44,16 @@ def _CreateInterfaceMap(self):
       dict, string MAC addresses mapped to the string network interface name.
     """
     interfaces = {}
-    for interface in os.listdir('/sys/class/net'):
+    for interface in netifaces.interfaces():
       try:
-        mac_address = open(
-            '/sys/class/net/%s/address' % interface).read().strip()
-      except (IOError, OSError) as e:
+        mac_address = (
+            netifaces.ifaddresses(interface)[netifaces.AF_LINK][0]['addr'])
```

Comment:
Yeah, understand. Exceptions in Python are slow so it's good to avoid them except for in exceptional cases (or when it's very inconvenient).